Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/ShapeToolIntegrationTest.java

```diff
@@ -79,44 +79,61 @@ public void setUp() throws Exception {
 	}
 
 	@Test
-	public void testRememberPositionAfterOrientationChange() {
+	public void testRememberShapeAfterOrientationChange() {
 		onView(withId(shape))
+				.perform(click())
+				.check(matches(isSelected()));
+		onToolBarView()
+				.performCloseToolOptions();
+
+		onView(isRoot())
 				.perform(click());
 
-		PointF p = new PointF(0, 0);
-		((BaseToolWithShape) PaintroidApplication.currentTool).toolPosition.set(p);
+		Bitmap expectedBitmap = PaintroidApplication.drawingSurface.getBitmapCopy();
+
+		onTopBarView()
+				.performUndo();
 
 		activityTestRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
 
-		assertEquals(p, ((BaseToolWithShape) PaintroidApplication.currentTool).toolPosition);
+		onView(withId(shape))
+				.check(matches(isSelected()));
+
+		onToolBarView()
+				.performCloseToolOptions();
+
+		onView(isRoot())
```

Comment:
Please click on the drawing surface. What purpose does the click have? If it's supposed to stamp the picture, use `onDrawingSurfaceView().perform(touchAt(DrawingSurfaceLocationProvider.TOOL_POSITION));`.